xbmc/video/VideoDatabase.cpp

```diff
@@ -5379,11 +5379,37 @@ void CVideoDatabase::UpdateTables(int iVersion)
 
   if (iVersion < 112)
     m_pDS->exec("ALTER TABLE settings ADD CenterMixLevel integer");
+
+  if (iVersion < 114)
+  {
+    m_pDS2->query("SELECT files.idFile, files.strFilename, path.strPath FROM files LEFT JOIN path ON files.idPath = path.idPath WHERE files.strFilename LIKE 'plugin://%'");
+    while (!m_pDS2->eof())
+    {
+      std::string path, fn;
+      SplitPath(m_pDS2->fv(1).get_asString(), path, fn);
+      if (path != m_pDS2->fv(2).get_asString())
+      {
+        int pathid = AddPath(path);
```

Comment:
It is not safe to call most other functions of this class that act upon the DB from the update code - this might break in the future because e.g. `AddPath` already expects the DB to be fully updated. Can you do it some other way? Moderate amounts of code duplicate are permissible and probably unavoidable in this case.

Also, as you probably noticed, `AddPath` uses `m_pDS` (which is why your code wouldn't work when you swap `m_pDS2` and `m_pDS` which would be more natural).